package.json

```diff
@@ -223,6 +223,7 @@
     "ngreact": "0.5.1",
     "node-fetch": "1.7.3",
     "node-forge": "^0.9.1",
+    "object-prototype": "^3.0.0",
```

Comment:
I think there's a lot of value in having code like this in a standalone module. If other people start using it outside of Kibana, we'll get a lot of extra eyes on the code that will benefit everybody. This is what happened with [`require-in-the-middle`](https://github.com/elastic/require-in-the-middle). If we had left that inside of [`elastic-apm-node`](https://github.com/elastic/apm-agent-nodejs), then it would not have been as good of a module today.

Another big upside is the ease of testing and maintenance. One of the big benefits of Node.js, in my opinion, is the module ecosystem and I really like the benefits you get from splitting code up into multiple modules that each have a single focus. By having a piece of code as a separate module, it forces you to really think of the API and treat the internals as a black box. In my experience, it also leads to better test coverage as you can create good unit tests really easily.

I know there are downsides to this approach as well - the main one being that updates to the module take a few extra steps to get into Kibana. But I think it's worth it in cases like this where the module is completely unrelated to Kibana.

For modules unrelated to our business cases, I personally prefer to not pollute the `elastic` GitHub org with the repo, but if you think it's best, I will of course move it 😃 

What do you think?

Reply:
Come to think of it, if I could have my way I'd probably also move the logic for patching the `child_process` module out into its own module on npm. That way we also didn't have to add `tape` as a dependency 😉